### PR TITLE
Changed loadIndexedBam to use hadoop-bam InputFormat

### PR DESCRIPTION
**Pull request by fnothaft**
A bit of cleanup on top of @andrewmchen's work in #953.

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/models/Interval.scala [hunk 1]
```diff
@@ -41,4 +42,19 @@ trait Interval {
    */
   def width: Long = end - start
 
+  /**
+   * Need to implement getStart function from Locatable. (1-based start position. closed)
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/models/Interval.scala]: Mixing 0- and 1- based coordinate systems in the same class gives me the hives. Can we hide this as an implementation detail in an adapter somewhere in the i/o code?

Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/models/Interval.scala]: I don't think we can hide it, but we could mark it as `@deprecated` to hint that it is only for use with legacy codebases?

Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/models/Interval.scala]: I guess I don't see where it is used, so I'm not sure what suggestion to make.  Will have to dig a bit.

Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/models/Interval.scala]: No, this allows us to supply `ReferenceRegion`s to the Hadoop-BAM indexed input format code without going through an intermediate class.

Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/models/Interval.scala]: Actually, think I may have misread.

> And is this the Interval that will need to go into bdg-utils to support IntervalRDD? Then there's no way I want to mix htsjdk stuff in.

@akmorrow13 @erictu, can you all comment?

Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/models/Interval.scala]: Actually, I get what you're saying now, @heuermh. I will refactor this out.

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala [hunk 2]
```diff
@@ -325,8 +325,7 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
    *        a single Bam file. The bam index file associated needs to have the same name.
    * @param viewRegion The ReferenceRegion we are filtering on
    */
-  def loadIndexedBam(
-    filePath: String, viewRegion: ReferenceRegion): RDD[AlignmentRecord] = {
+  def loadIndexedBam(filePath: String, viewRegion: ReferenceRegion): RDD[AlignmentRecord] = {
```
Comment (erictu) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala]: Should this use AlignmentRecordRDD?

**Comment by fnothaft** [comment]
Rebased and addressed comments. Ping @heuermh @erictu for review.

File: adam-core/src/main/java/org/bdgenomics/adam/io/IndexedBamInputFormat.scala [hunk 3]
```diff
@@ -1,102 +0,0 @@
...
-        val end: Long = chunk.getChunkEnd()
-        val locs = Array[String]()
-        val newSplit = new FileVirtualSplit(filePath, start, end, locs)
-        splits += newSplit
-      }
-      splits.toList
-    }
-  }
-
-}
```
Comment (heuermh) [inline adam-core/src/main/java/org/bdgenomics/adam/io/IndexedBamInputFormat.scala]: oh how I like to see code being removed  :)

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala [hunk 4]
```diff
@@ -63,6 +64,13 @@ import scala.collection.JavaConversions._
 import scala.collection.Map
 import scala.reflect.ClassTag
 
+// only used with indexedbamload
+private case class LocatableReferenceRegion(rr: ReferenceRegion) extends Locatable {
```
Comment (heuermh) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala]: perfect

**Comment by heuermh** [comment]
LGTM

File: adam-core/src/main/java/org/bdgenomics/adam/io/IndexedBamInputFormat.scala (hunk 3 above)
Comment (fnothaft) [inline adam-core/src/main/java/org/bdgenomics/adam/io/IndexedBamInputFormat.scala]: Indeed; it is uniquely satisfying!

**Comment by erictu** [comment]
+1

**Comment by fnothaft** [comment]
Rebased and ready to merge!

**Comment by heuermh** [comment]
Oops, didn't let Jenkins have a final say before merging.  Builds & tests ok for me locally.

Thank you, @fnothaft @andrewmchen!

**Comment by fnothaft** [comment]
Looks like the @AmplabJenkins failures were just the git `failed to determine` issue. We should be a-OK but I'll follow up if the ToT build fails.

**Pull request merged**